python/tvm/relay/op/transform.py

```diff
@@ -242,3 +243,35 @@ def where(condition, x, y):
     Note that the shape of condition, x, and y needs to be the same.
     """
     return _make.where(condition, x, y)
+
+
+def split(data, indices_or_sections, axis=0, equal_split=False):
```

Comment:
as per numpy convention, indices_or_sections
 
- can either be an integer(which indicates equal split)
-  or Array of Int, which(indicate the sections)

We want the API behavior to be as consistent as possible. In order to do so, we can store the indices_or_sections as NodeRef in the Attrs.  Now, in the _make.split C++ API, we should check if the input type is kDLInt, and convert that to IntImm.

During type inference, we can check if the indices_or_sections is IntImm, and decide accordingly.

Reply:
This does make the code a bit harder to read and we should explain clearly by commenting on the field of indices_or_sections in the Attrs, to say the two expected field, and the reason of doing so(compatible with numpy convention). We will later try to introduce a Union type which might make it more readable.